tools/wptrunner/wptrunner/testdriver-extra.js

```diff
@@ -1,39 +1,85 @@
 "use strict";
 
-(function(){
-    let pending_resolve = null;
-    let pending_reject = null;
+(function() {
+    const is_test_context = window.__wptrunner_message_queue !== undefined;
+    const pending = new Map();
+    const super_set_context = window.test_driver.set_test_context;
```

Comment:
This appears to only be used once; was there a reason to have it as a 'constant' here rather than just call `window.test_driver.set_test_context` in the overriden `set_test_context`?

Reply:
It's not a real override, it actually redefines the function on window. I've changed it around so that it uses the normal thing of calling into _internal, but the order is dfferent from normal.